src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.Variables.cs

```diff
@@ -0,0 +1,373 @@
+                {
+                    _variableBySlot.Add(default);
+                }
+                _variableBySlot[index] = identifier;
+                return ConstructSlot(Id, index);
+            }
+
+            internal bool TryGetType(Symbol symbol, out TypeWithAnnotations type)
+            {
+                var variables = GetVariablesContainingVariable(symbol);
+                return variables._variableTypes.TryGetValue(symbol, out type);
+            }
+
+            internal void SetType(Symbol symbol, TypeWithAnnotations type)
+            {
+                var variables = GetVariablesContainingVariable(symbol);
+                Debug.Assert((object)variables == this);
+                variables._variableTypes[symbol] = type;
+            }
+
+            internal VariableIdentifier this[int slot]
+            {
+                get
+                {
+                    (int id, int index) = DeconstructSlot(slot);
+                    var variables = GetVariablesForId(id);
+                    return variables!._variableBySlot[index];
+                }
+            }
+
+            internal int Count => _variableBySlot.Count;
+
+            internal void GetMembers(ArrayBuilder<(VariableIdentifier, int)> builder, int containingSlot)
+            {
+                (int id, int index) = DeconstructSlot(containingSlot);
+                var variables = GetVariablesForId(id)!;
+                var variableBySlot = variables._variableBySlot;
+                for (index++; index < variableBySlot.Count; index++)
+                {
+                    var variable = variableBySlot[index];
+                    if (variable.ContainingSlot == containingSlot)
+                    {
+                        builder.Add((variable, ConstructSlot(id, index)));
+                    }
+                }
+            }
+
+            private Variables GetVariablesForVariable(VariableIdentifier identifier)
+            {
+                int containingSlot = identifier.ContainingSlot;
+                if (containingSlot > 0)
+                {
+                    return GetVariablesForId(DeconstructSlot(containingSlot).Id)!;
+                }
+                return GetVariablesContainingVariable(identifier.Symbol);
+            }
+
+            internal Variables GetVariablesContainingVariable(Symbol symbol)
+            {
+                switch (symbol)
+                {
+                    case LocalSymbol:
+                    case ParameterSymbol:
+                    case MethodSymbol { MethodKind: MethodKind.LocalFunction }:
+                        if (symbol.ContainingSymbol is MethodSymbol method &&
+                            GetVariablesForSymbol(method.PartialImplementationPart ?? method) is { } variables)
+                        {
+                            return variables;
+                        }
+                        break;
+                }
+                return GetVariablesRoot();
```

Comment:
>return GetVariablesRoot(); [](start = 16, length = 26)

In what cases do we get here?

Reply:
> In what cases do we get here?

Static fields and variables declared in field initializers.

---
In reply to: [561458311](https://github.com/dotnet/roslyn/pull/50417#discussion_r561458311) [](ancestors = 561458311)

Reply:
>> In what cases do we get here?

> Static fields and variables declared in field initializers.

Consider adding a comment.

---
In reply to: [561602717](https://github.com/dotnet/roslyn/pull/50417#discussion_r561602717) [](ancestors = 561602717,561458311)